internal/handler/dynv6/dynv6_handler.go

```diff
@@ -0,0 +1,116 @@
+package dynv6
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"runtime/debug"
+	"strings"
+	"time"
+
+	"github.com/TimothyYe/godns/internal/settings"
+	"github.com/TimothyYe/godns/internal/utils"
+	"github.com/TimothyYe/godns/pkg/notify"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	// Dynv6Url the API address for Duck DNS
+	Dynv6Url = "https://dynv6.com/api/update?hostname=%s&token=%s&%s"
+)
+
+// Handler struct
+type Handler struct {
+	Configuration *settings.Settings
+}
+
+// SetConfiguration pass dns settings and store it to handler instance
+func (handler *Handler) SetConfiguration(conf *settings.Settings) {
+	handler.Configuration = conf
+}
+
+// DomainLoop the main logic loop
+func (handler *Handler) DomainLoop(domain *settings.Domain, panicChan chan<- settings.Domain) {
+	defer func() {
+		if err := recover(); err != nil {
+			log.Errorf("Recovered in %v: %v\n", err, string(debug.Stack()))
+			panicChan <- *domain
+		}
+	}()
+
+	looping := false
+
+	for {
+		if looping {
+			// Sleep with interval
+			log.Debugf("Going to sleep, will start next checking in %d seconds...", handler.Configuration.Interval)
+			time.Sleep(time.Second * time.Duration(handler.Configuration.Interval))
+		}
+
+		looping = true
+		currentIP, err := utils.GetCurrentIP(handler.Configuration)
+
+		if err != nil {
+			log.Error("get_currentIP:", err)
+			continue
```

Comment:
Will it cause an infinite loop here once the program failed to get the current IP address?